src/script/connection/ConnectionRepository.ts

```diff
@@ -187,7 +185,9 @@ export class ConnectionRepository {
    * @returns User connection entity
    */
   getConnectionByConversationId(conversationId: string): ConnectionEntity {
-    return this.connectionEntities().find(connectionEntity => connectionEntity.conversationId === conversationId);
+    const connectionEntities = Object.values(this.connectionEntities());
+    const match = connectionEntities.find(connectionEntity => connectionEntity.conversationId === conversationId);
+    return match ? this.getConnectionByUserId(match.userId) : undefined;
```

Comment:
Can the match be returned directly here?

Reply:
No because it will return a copy and not the original observable. We need to guarantee identity here.